components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediatorFactory.java

```diff
@@ -202,4 +181,161 @@ private List extractOperations(OMElement operationsElemet, DataServiceCallMediat
+                    }
+                    String operationType = operationsTypeElement.getAttributeValue(new QName(DataServiceCallMediatorConstants.TYPE));
+                    if (operationType == null) {
+                        handleException("The 'type' attribute in 'operations' element  is missing in the configuration.");
+                    }
+                    OperationsType operationsType = OperationsType.valueOf(operationType.toUpperCase());
+                    List operationList = extractOperations(operationsTypeElement, mediator);
+                    if (OperationsType.SINGLE_REQ.equals(operationsType) && operationList.size() > 1) {
+                        handleException("The 'single operation' should contain one operation in the configuration.");
+                    }
+                    Operations operations = mediator.new Operations(operationsType, operationList);
+                    mediator.setOperations(operations);
+                } else if (sourceType.equals(DataServiceCallMediatorConstants.PROPERTY)) {
+                    OMAttribute propertyAtr = sourceElement.getAttribute(NAME_Q);
+                    if (propertyAtr != null) {
+                        if (!propertyAtr.getAttributeValue().isEmpty()) {
+                            String propertyName = propertyAtr.getAttributeValue();
+                            if (isDynamicName(propertyName)) {
+                                Value sourceDynamicName = createDynamicNameValue(propertyName, sourceElement);
+                                mediator.setSourceDynamicName(sourceDynamicName);
+                            } else {
+                                mediator.setSourcePropertyName(propertyName);
+                            }
+                        } else {
+                            handleException("The 'name' attribute in 'source' element is empty. " +
+                                    "Please enter a value.");
+                        }
+                    } else {
+                        handleException("The 'name' attribute in 'source' element  is missing " +
+                                "in the configuration.");
+                    }
+                } else {
+                    if (operationsTypeElement != null) {
+                        handleException("The 'source' type is set to body. Inline configurations " +
+                                "are only applicable for source type 'inline'.");
+                    }
+                }
+            } else {
+                handleException("The 'type' attribute in 'source' element is required for the configuration");
+            }
+        } else {
+            log.warn("The 'source' element is missing in the configuration. The default value will be set to 'body'.");
+            if (operationsTypeElement != null) {
+                handleException("The 'source' type is set to body. Inline configurations are " +
```

Comment:
With hanlde exception method I guess the deployment breaks.. So the default value doesn't have any effect and it will become a faulty artifact.

Reply:
Fixed